source/gitlab/gitlab.go

```diff
@@ -120,8 +132,16 @@ func WithInstance(client *gitlab.Client, config *Config) (source.Driver, error)
 func (g *Gitlab) readDirectory() error {
 	var nodes []*gitlab.TreeNode
 
+	g.listOptions = &gitlab.ListTreeOptions{
```

Comment:
`readDirectory()` shouldn't modify the receiver e.g. `g *Gitlab`
Instead, the Gitlab instance should be configured upfront when it's created